File: backend/internal/apis/dtos/stream.go
package dtos

type StreamResponse struct {
	Event string      `json:"event"` // ai-response, ai-response-step, ai-response-error, db-connected, db-disconnected, sse-connected, response-cancelled, query-results, rollback-executed, rollback-query-failed
	Data  interface{} `json:"data,omitempty"`
}

File: client/src/types/stream.ts
export interface StreamResponse {
    event: 'ai-response' | 'ai-response-step' | 'ai-response-error' | 'db-connected' |
    'db-disconnected' | 'sse-connected' | 'response-cancelled' | 'query-results' |
    'rollback-executed' | 'query-execution-failed' | 'rollback-query-failed';
    data?: any;
} 